Realm.Shared/Handles/ObjectHandle.cs

```diff
@@ -127,8 +127,13 @@ private static class NativeMethods
 
             [DllImport(InteropConfig.DLL_NAME, EntryPoint = "object_remove_row", CallingConvention = CallingConvention.Cdecl)]
             public static extern void remove_row(ObjectHandle handle, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "object_equals_object", CallingConvention = CallingConvention.Cdecl)]
+            public static extern IntPtr equals_object(ObjectHandle handle, ObjectHandle otherHandle, out NativeException ex);
```

Comment:
This should return a `bool` instead of `IntPtr`. However, because the default size of the managed boolean type in .NET is different than the size of the C++ `bool` type, you will need to apply `[return: MarshalAs(UnmanagedType.I1)]`.